examples/with-stacks/src/index.ts

```diff
@@ -0,0 +1,156 @@
+import { Turnkey as TurnkeyServerSDK } from "@turnkey/sdk-server";
+import {
+  broadcastTransaction,
+  createMessageSignature,
+  createStacksPublicKey,
+  leftPadHex,
+  makeUnsignedContractCall,
+  PubKeyEncoding,
+  publicKeyIsCompressed,
+  sigHashPreSign,
+  SingleSigSpendingCondition,
+  TransactionSigner,
+  txidFromData,
+  type StacksTransactionWire,
+} from "@stacks/transactions";
+import { hexToBytes } from "@stacks/common";
+import * as path from "path";
+import * as dotenv from "dotenv";
+
+// Load environment variables from `.env.local`
+dotenv.config({ path: path.resolve(process.cwd(), ".env.local") });
+
+// Define the Turnkey API client
+const client = new TurnkeyServerSDK({
+  apiBaseUrl: process.env.TURNKEY_BASE_URL!,
+  apiPrivateKey: process.env.TURNKEY_API_PRIVATE_KEY!,
+  apiPublicKey: process.env.TURNKEY_API_PUBLIC_KEY!,
+  defaultOrganizationId: process.env.TURNKEY_ORGANIZATION_ID!,
+});
+
+const constructStacksTx = async (pubKey: string) => {
+  let transaction = await makeUnsignedContractCall({
+    contractAddress: "SP3TJMRQ13QR6V5HGT6AKEK7PP699F4148JZTB9G3",
+    contractName: "counter",
+    functionName: "increment",
+    functionArgs: [],
+    numSignatures: 1,
+    publicKey: pubKey,
+    postConditions: [],
+    postConditionMode: "deny",
+    network: "mainnet",
+  });
+
+  // The `signer` contains the `sigHash` property needed for the `preSignSigHash`
+  const signer = new TransactionSigner(transaction);
+
+  return { stacksTransaction: transaction, stacksTxSigner: signer };
+};
+
+const generatePreSignSigHash = (
+  transaction: StacksTransactionWire,
+  signer: TransactionSigner,
+) => {
+  let preSignSigHash = sigHashPreSign(
+    signer.sigHash,
+    transaction.auth.authType,
+    transaction.auth.spendingCondition.fee,
+    transaction.auth.spendingCondition.nonce,
+  );
+
+  return preSignSigHash;
+};
+
+const generatePostSignSigHash = (
+  pubKey: string,
+  preSignSigHash: string,
+  nextSig: string,
+) => {
+  const RECOVERABLE_ECDSA_SIG_LENGTH_BYTES = 65;
+  const hashLength = 32 + 1 + RECOVERABLE_ECDSA_SIG_LENGTH_BYTES;
+
+  let pubKeyStacksWire = createStacksPublicKey(pubKey);
+
+  const pubKeyEncoding = publicKeyIsCompressed(pubKeyStacksWire.data)
+    ? PubKeyEncoding.Compressed
+    : PubKeyEncoding.Uncompressed;
+
+  const sigHash =
+    preSignSigHash + leftPadHex(pubKeyEncoding.toString(16)) + nextSig;
+  const sigHashBytes = hexToBytes(sigHash);
+
+  if (sigHashBytes.byteLength > hashLength) {
+    throw Error("Invalid signature hash length");
+  }
+
+  let nextSigHash = txidFromData(sigHashBytes);
+
+  return nextSigHash;
+};
+
+const signStacksTx = async () => {
+  try {
+    const stacksPublicKey =
```

Comment:
same here, I think this public key should come from an env var if we think people running this example will need to override it